### PR TITLE
Export PSBT UR as untagged CBOR

### PR DESCRIPTION
**Pull request by praveenperera**
## Summary by CodeRabbit

* **Bug Fixes**
  * Updated PSBT encoding format to improve compatibility while maintaining backward compatibility with previously encoded transactions.

* **Tests**
  * Added comprehensive unit test coverage for PSBT encoding and decoding across various scenarios, including single and multi-part transactions, edge cases, and malformed input handling.

* **Chores**
  * Updated tooling dependencies to align with new encoding approach.

### CODE REVIEW

File: rust/crates/cove-ur/src/crypto_psbt.rs [hunk 1]
```diff
@@ -63,26 +63,20 @@ impl CryptoPsbt {
...
         let mut encoder = Encoder::new(&mut buffer);
 
-        // write tag 310
-        encoder.tag(Tag::new(CRYPTO_PSBT)).map_err_str(UrError::CborEncodeError)?;
-
-        // write PSBT as byte string
         encoder.bytes(&psbt_bytes).map_err_str(UrError::CborEncodeError)?;
 
         Ok(buffer)
     }
```
[the remark this answers is not in the document]
Comment (praveenperera) [reply rust/crates/cove-ur/src/crypto_psbt.rs:72]: Current BCR-2020-006 treats the top-level UR type as the type marker and defines the PSBT payload as plain CBOR bytes; tags are for nested CBOR contexts. The existing Jade vector also decodes as a byte string payload rather than a top-level #6.310(bytes) tag. Decoding still accepts both tagged and untagged inputs for compatibility.
